### PR TITLE
Create index API from Python

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

**Comment by eddyxu** [comment]
Verified it works, but caveats is that it needs to build lance with `faiss` feature.

I will work on #471 to remove faiss dependency.

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -288,6 +288,31 @@ def versions(self):
...
+        Parameters
+        ----------
+        column : str
+            The column to be indexed.
+        index_type : str
+            The type of the index. Only support "IVF_PQ" now.
+        name : str, optional
+            The index name. If not provided, an index name will be generated from the
+            column name.
+        kwargs :
```
Comment (changhiskhan) [inline python/python/lance/dataset.py:307]: can you explicitly document the num partitions and num subvectors parameters?
also 1) indicate what the default values are and 2) add a link to those papers maybe?

File: python/src/dataset.rs [hunk 2]
```diff
@@ -180,6 +183,40 @@ impl Dataset {
...
     }
+
+    fn create_index(
+        self_: PyRef<'_, Self>,
+        columns: Vec<&str>,
+        index_type: &str,
+        name: Option<String>,
+        kwargs: &PyDict,
+    ) -> PyResult<()> {
+        let idx_type = match index_type.to_uppercase().as_str() {
```
Comment (changhiskhan) [inline python/src/dataset.rs:194]: do we need to add a check that the vector column is fixed size array of float32?

**Review by changhiskhan — APPROVED** [review]
two quick questions

File: python/src/dataset.rs (hunk 2 above)
Comment (eddyxu) [reply python/src/dataset.rs:194]: The check was done in the core `/rust` side.

File: python/python/lance/dataset.py (hunk 1 above)
Comment (eddyxu) [reply python/python/lance/dataset.py:307]: Ok, will do.

**Pull request merged**